### PR TITLE
Core/#8264 room types creation ui

### LINKED ISSUE #8264: [BUG] [UI] Sidenav selected color not readable
Hi guys,
Sidebar selected color not readable. I think it's related to #8255 , #8257  See below gif:
peek 2017-09-23 09-37

**/CC:** @ggazzo , @rodrigok

### PR DESCRIPTION
**Pull request by mrsimpson**
# Purpose 

@RocketChat/core as discussed with @graywolf336, the feature for adding custom room types has been split into two parts: #8274 adds the backend and API in order to create custom room types.
This PR adds to it a tabbed UI-Element. 

Closes #8264 - adds the UI-part based upon #8274 

#8274 has already been merged into this branch, so it looks bigger than it actually is. Given that the backend-part has already been approved, you can compare the actual addition by [comparing the two branches](https://github.com/assistify/Rocket.Chat/compare/core/%238264-room-types...assistify:core/%238264-room-types-creation-ui)

# UI

This is how it looks like:
tabbed creation dialog
If no custom room types are added, the UI is unchanged.

# What's missing
What I couldn't do due to lack of knowledge

- [x] Create Unit-Test adding a custom room type
- [ ]  Find a better responsive solution than enabling scrolling - but this is actually not an issue of the tabs, but of 0.59 as a whole (see #8603)

### CODE REVIEW

**Comment by cyclops24** [comment]
@mrsimpson Can you tell me your username on `open.rocket.chat`?

**Comment by geekgonecrazy** [comment]
@mrsimpson this is also great :grin: I might have missed it.  But is there a way to prevent a room type from being added on this screen?

File: packages/rocketchat-ui/client/views/app/createRooms.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+const roomTypesAfterStandard = function() {
+	const orderHigh = RocketChat.roomTypes.roomTypesOrder.filter((roomTypeOrder) => roomTypeOrder.identifier === 'd')[0].order;
+	return RocketChat.roomTypes.roomTypesOrder.filter(
+		(roomTypeOrder) => roomTypeOrder.order > orderHigh
+	).map(
+		(roomTypeOrder) => {
+			return RocketChat.roomTypes.roomTypes[roomTypeOrder.identifier];
+		}
+	).filter((roomType) => roomType.creationTemplate && roomType.canBeCreated());
```
Comment (mrsimpson) [inline packages/rocketchat-ui/client/views/app/createRooms.js]: @geekgonecrazy There are actually two ways of preventing the room type from creating a tab: Only those with a creation template are being included and for each of them, the validation for the permissions is performed.

Comment (geekgonecrazy) [reply packages/rocketchat-ui/client/views/app/createRooms.js]: Perfect :+1:

**Comment by mrsimpson** [comment]
since no one cared about this for so long we decided to abandon the solution. Implementing custom room types and working upstream seems not to be possible.

**Comment by mnlbox** [comment]
Oh man. It's very cool feature :disappointed:

**Pull request closed without merging**